src/generator/models.ts

```diff
@@ -132,6 +132,16 @@ class StructDef {
     if (this.Language.errorType) {
       text += '\traw string\n';
     }
+    // group fields by required/optional/read-only in that order
```

Comment:
Just confirming the below if is also checking in this order?

Reply:
Correct.  The properties are already lexicographically sorted.  This sort groups them by their required/optional/read-only status.